tests/std/include/range_algorithm_support.hpp

```diff
@@ -449,78 +503,195 @@ template <class I1, class I2>
+        Continuation::template call<Args..., range<random, int, true, true, false, true, false>>();
+        Continuation::template call<Args..., range<random, int, true, true, false, true, true>>();
+        Continuation::template call<Args..., range<random, int, true, true, true, true, false>>();
+        Continuation::template call<Args..., range<random, int, true, true, true, true, true>>();
+
+        // Ditto always CommonIterators; !Sized && SizedSentinel is uninteresting (ranges::size still works), as is
+        // !Sized && Common. contiguous also implies !Proxy.
+        Continuation::template call<Args..., range<contiguous, int, false, false, false, true, false>>();
+        Continuation::template call<Args..., range<contiguous, int, true, false, false, true, false>>();
+        Continuation::template call<Args..., range<contiguous, int, true, false, true, true, false>>();
+        Continuation::template call<Args..., range<contiguous, int, true, true, false, true, false>>();
+        Continuation::template call<Args..., range<contiguous, int, true, true, true, true, false>>();
     }
 };
 
 template <class Continuation>
 struct with_forward_ranges {
     template <class... Args>
     static void call() {
-        Continuation::template call<Args..., test_range<std::forward_iterator_tag, int, false, false>>();
-        Continuation::template call<Args..., test_range<std::forward_iterator_tag, int, false, true>>();
-        Continuation::template call<Args..., test_range<std::forward_iterator_tag, int, true, false>>();
-        Continuation::template call<Args..., test_range<std::forward_iterator_tag, int, true, true>>();
-
-        Continuation::template call<Args..., test_range<std::bidirectional_iterator_tag, int, false, false>>();
-        Continuation::template call<Args..., test_range<std::bidirectional_iterator_tag, int, false, true>>();
-        Continuation::template call<Args..., test_range<std::bidirectional_iterator_tag, int, true, false>>();
-        Continuation::template call<Args..., test_range<std::bidirectional_iterator_tag, int, true, true>>();
-
-        Continuation::template call<Args..., test_range<std::random_access_iterator_tag, int, true, false>>();
-        Continuation::template call<Args..., test_range<std::random_access_iterator_tag, int, true, true>>();
-
-        Continuation::template call<Args..., test_range<std::contiguous_iterator_tag, int, true, false>>();
-        Continuation::template call<Args..., test_range<std::contiguous_iterator_tag, int, true, true>>();
+        using namespace test;
+
+        // forward always has CommonIterators; !Sized && SizedSentinel is uninteresting (sized_range is sized_range).
+        Continuation::template call<Args..., range<fwd, int, false, false, false, true, false>>();
+        Continuation::template call<Args..., range<fwd, int, false, false, false, true, true>>();

```

Comment:
Occurs in multiple files: These bool parameters are effectively unreadable. Would it be better to have `enum class` enumerators?

Reply:
Yes, this has been bothering me. I'm concerned that longer parameter names will make it harder to see which permutations have been omitted from lists like this - `0` and `1` would be even better than `true` and `false`. I'll poke around a bit and see if I can solve both problems.

Reply:
I kept these `enum class` changes separate from the others in the two new commits - I'm not terribly enthused, but I'm too close to this to be objective.